File: src/main/java/io/github/shirohoo/library/persistance/book/BookRepositoryImpl.java
package io.github.shirohoo.library.persistance.book;

import io.github.shirohoo.library.domain.book.Book;
import io.github.shirohoo.library.domain.book.BookRepository;
import java.util.Optional;
import org.springframework.stereotype.Repository;

@Repository
public class BookRepositoryImpl implements BookRepository {
    private final BookJpaRepository jpaRepository;

    public BookRepositoryImpl(BookJpaRepository jpaRepository) {
        this.jpaRepository = jpaRepository;
    }

    @Override
    public Book save(Book newBook) {
        if (newBook == null) {
            return null;
        }
        return jpaRepository.save(newBook);
    }

    @Override
    public Optional<Book> findBy(String bookTitle) {
        return jpaRepository.findByTitle(bookTitle);
    }
}

File: src/main/kotlin/io/github/shirohoo/library/application/persistence/book/BookRepositoryImpl.kt
package io.github.shirohoo.library.application.persistence.book

import io.github.shirohoo.library.domain.book.Book
import io.github.shirohoo.library.domain.book.BookRepository
import org.springframework.stereotype.Repository

@Repository
class BookRepositoryImpl(
    private val jpaRepository: BookJpaRepository
) : BookRepository {
    override fun save(newBook: Book): Book {
        return jpaRepository.save(newBook)
    }

    override fun findBy(bookTitle: String): Book {
        return jpaRepository.findByTitle(bookTitle)
            .orElseThrow { NoSuchElementException("book not found.") }
    }
}